File: exam-slammr-webapp/src/main/java/uk/co/claritysoftware/exam/slammr/webapp/config/ApplicationConfig.java
package uk.co.claritysoftware.exam.slammr.webapp.config;

import java.util.Optional;
import org.springframework.beans.factory.annotation.Value;
import org.springframework.stereotype.Component;

import lombok.Getter;

/**
 * Simple bean encapsulating the application configuration properties
 */
@Component
@Getter
public class ApplicationConfig {

	@Value("${aws.region}")
	private String awsRegion;

	@Value("${aws.accessKeyId}")
	private String awsAccessKeyId;

	@Value("${aws.secretKey}")
	private String awsSecretKey;

	@Value("${aws.dynamodb.endpoint:#{null}}")
	private Optional<String> awsDynamoDbEndpoint;

}

File: exam-slammr-webapp/src/main/kotlin/uk/co/claritysoftware/exam/slammr/webapp/config/ApplicationConfig.kt
package uk.co.claritysoftware.exam.slammr.webapp.config

import org.springframework.beans.factory.annotation.Value
import org.springframework.stereotype.Component
import java.util.*

/**
 * Simple bean encapsulating the application configuration properties
 */
@Component
class ApplicationConfig {

  @Value("\${aws.region}")
  val awsRegion: String? = null

  @Value("\${aws.accessKeyId}")
  val awsAccessKeyId: String? = null

  @Value("\${aws.secretKey}")
  val awsSecretKey: String? = null

  @Value("\${aws.dynamodb.endpoint:#{null}}")
  val awsDynamoDbEndpoint: Optional<String>? = null

}
